lib/dgram.js

```diff
@@ -272,6 +267,41 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+Socket.prototype.send = function(buffer,
+                                 offset,
+                                 length,
+                                 port,
+                                 address,
+                                 callback) {
+  var self = this;
+
+  if (arguments.length === 5 || arguments.length === 6) {
+    buffer = sliceBuffer(buffer, offset, length);
+  }
+
+  if (arguments.length < 5) {
+    callback = port;
+    port = offset;
+    address = length;
+  }
+
+  if (!Array.isArray(buffer))
+    buffer = [buffer];
+
+  length = 0;
+
+  for (var i = 0, l = buffer.length; i < l; i++) {
+    if (typeof buffer[i] === 'string')
```

Comment:
dgram supports strings without encoding at this point. I'm up for supporting only Buffers in this new API.
Strings without encoding are just "bad".

Original: https://github.com/nodejs/node/blob/master/lib/dgram.js#L253-L254